src/services/services.ts

```diff
@@ -31,6 +31,8 @@ namespace ts {
     /** The version of the language service API */
     export const servicesVersion = "0.5";
 
+    const ruleProvider: formatting.RulesProvider = new formatting.RulesProvider();

```

Comment:
this will force users of the services, even ones who never use formatting, to allocate this large data structure. how about doing on first request instead?